From: Mark Michelson <mmichelson@digium.com>
Subject: Be more consistent with the return code for requests received from invalid domain.
 When Asterisk receives an INVITE from an external domain when allowexternaldomains=no
 send a 403 instead of a 404. This is consistent with Asterisk's behavior when receiving
 a REGISTER in this situation.
Origin: upstream, http://svnview.digium.com/svn/asterisk?view=revision&revision=369302
Bug: https://issues.asterisk.org/jira/browse/ASTERISK-19601
--- a/channels/chan_sip.c
+++ b/channels/chan_sip.c
@@ -22091,8 +22091,6 @@ static int handle_request_invite(struct
 				}
 			/* INTENTIONAL FALL THROUGH */
 			case SIP_GET_DEST_EXTEN_NOT_FOUND:
-			case SIP_GET_DEST_REFUSED:
-			default:
 				{
 					char *decoded_exten = ast_strdupa(p->exten);
 					transmit_response_reliable(p, "404 Not Found", req);
@@ -22101,6 +22099,10 @@ static int handle_request_invite(struct
 						" '%s' rejected because extension not found in context '%s'.\n",
 						S_OR(p->username, p->peername), ast_sockaddr_stringify(&p->recv), decoded_exten, p->context);
 				}
+				break;
+			case SIP_GET_DEST_REFUSED:
+			default:
+				transmit_response_reliable(p, "403 Forbidden", req);
 			} /* end switch */
 
 			p->invitestate = INV_COMPLETED;
